### PR TITLE
Migrate the transaction logger to V2 GRPC API.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Migrate the transaction logger to V2 GRPC API.
This simplifies some of the waiting code since the update Rust SDK has methods to wait for new finalized blocks.

## Changes

- Bump minimum rust version to 1.61 to deal with new dependency versions.
- Make the necessary changes to the code to accommodate the new API.

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -346,128 +320,111 @@ enum NodeError {
...
+        let r = accounts
+            .try_fold(HashSet::new(), |mut cc, addr| async move {
+                let _ = cc.insert(AccountAddressEq(addr));
+                Ok(cc)
+            })
+            .await?;
+        *canonical_cache = r;
     }
-    let mut parallel = 1;
+    let mut stream = node.get_finalized_blocks_from(*height).await?;
```
Comment (limemloh) [inline src/main.rs:349]: Consider provider a more detailed name instead of the type.
maybe `finalized_blocks`

File: src/main.rs [hunk 2]
```diff
@@ -346,128 +320,111 @@ enum NodeError {
...
+                    .response
+                    .try_collect()
+                    .await?;
+                Ok::<
+                    (BlockInfo, Vec<BlockItemSummary>, Vec<SpecialTransactionOutcome>),
+                    anyhow::Error,
+                >((binfo.response, events, special))
+            }
+        });
         for result in futures::future::join_all(handles).await {
```
Comment (limemloh) [inline src/main.rs:381]: We could optimize this a bit by processing as soon as something is ready, using either [FuturesOrdered](https://docs.rs/futures/latest/futures/stream/struct.FuturesOrdered.html) or [FuturesUnordered](https://docs.rs/futures/latest/futures/stream/struct.FuturesUnordered.html), but maybe out of scope of this PR

**Review by limemloh — APPROVED** [review]
LGTM

File: src/main.rs (hunk 2 above)
Comment (abizjak) [reply src/main.rs:381]: I changed it now to use futuresordered now, but only replacing the join_all.

The other change would be more complex.

**Pull request merged**